202 add input dropdown label to forms scss file

ashberg opened the pull request:
Fixes #202.

ashberg committed:
Working on adding classes to forms.scss.

ashberg committed:
Added the instructor_files and expected_student_files fields to all Project instances.

ashberg committed:
The new_course_semester is now chosen using a select element.

ashberg committed:
Added the dropdown_height prop to the cloning_destinations_dropdown in the single_project component.

james-perretta commented on src/styles/forms.scss:
```diff
@@ -8,3 +10,113 @@
+
+// Organization -----------------------------------
+
+legend {
```
I'd prefer to use classes for our custom form styles rather than specifying tags. Classes make it clearer where applied styling is coming from rather than having it applied "globally".

ashberg committed:
Changed fieldset, legend, and input[type=checkbox] in forms.scss to .fieldset, .legend, and .checkbox.

james-perretta commented on src/components/course_admin/manage_projects/single_project.vue:
```diff
@@ -49,16 +49,17 @@
                 <div tabindex="1" class="dropdown-header-wrapper">
                   <div id="input-course-to-copy-to" class="dropdown-header">
-                    {{course_to_clone_to.name ? course_to_clone_to.name : ""}}
+                    {{course_to_clone_to.name ? course_to_clone_to.name : ""}} -
```
I think the `-` should only be there if there semester is not null?

james-perretta commented on src/components/course_admin/manage_projects/single_project.vue:
```diff
@@ -67,7 +68,7 @@
                 <div slot-scope="{item}">
                   <span>
-                    {{item.name ? item.name : ""}}
+                    {{item.name ? item.name : ""}} -
```
Same here. It's probably worth writing a utility function that takes in a course and returns a string with name, semester, year

james-perretta commented on src/styles/forms.scss:
```diff
@@ -1,10 +1,136 @@
+    color: #ced4da;
+    content: '\f0c8';
+    display: inline-block;
+    font-family: "Font Awesome 5 Free";
```
I'm not crazy about the way the checkboxes look. Just having a gray square doesn't really look like a checkbox for me. 
I'd be in support of either the default way checkboxes look or some kind of rectangle with a black border and a white center (like the way they look on google forms).

james-perretta reviewed (changes requested):
(Also see comment about course name/semester/year displaying)

james-perretta reviewed (changes requested):
(Oops, double post)

ashberg committed:
Changed the color of the outline of the checkbox square to be darker.

james-perretta reviewed (changes requested):
It looks like disabled checkboxes still have the gray square look (whether checked or not). Ideally, we want the box to still look checked if it was checked before becoming disabled (if it's checked and disabled, it should look checked but grayed out). 

You might also take this opportunity to remove some of the duplication between `.checkbox + label::before` and `.checkbox:disabled + label::before`.

ashberg committed:
Added a get_course_info function to utils.ts.

ashberg committed:
Removed the line that checked if course.name was null from get_course_info.

ashberg committed:
Changed the styling of checkboxes in forms.scss.

ashberg committed:
Reduced the border-radius and padding on the checkbox element.

james-perretta committed:
Reduced margin between checkbox and label.
Renamed get_course_info to format_course_name.
Removed extra spaces produced by format_course_name when semester or year is null.
